pkg/commands/workload.go

```diff
@@ -771,7 +772,47 @@ func isUrl(str string) (bool, error) {
 	}
 }
 
-func (opts *WorkloadOptions) GetReadyConditionWorkers(c *cli.Config, workload *cartov1alpha1.Workload, workers []wait.Worker) []wait.Worker {
+func (opts *WorkloadOptions) RaceWithTimeout(ctx context.Context, c *cli.Config, shouldPrint bool, workers []wait.Worker) error {
+	err := wait.Race(ctx, opts.WaitTimeout, workers)
+	// print wait error only if output is not set or it was not used with --yes
+	if err != nil {
+		if err == context.DeadlineExceeded {
+			cli.PrintPrompt(shouldPrint, c.Printf, "%s timeout after %s waiting for %q to become ready\n", printer.Serrorf("Error:"), opts.WaitTimeout, opts.Name)
+		} else {
+			cli.PrintPrompt(shouldPrint, c.Eprintf, "%s %s\n", printer.Serrorf("Error:"), err)
+		}
+	}
+	return err
+}
+
+func GetStatusChangeWorkers(c *cli.Config, workload *cartov1alpha1.Workload, currentWorkload *cartov1alpha1.Workload, workers []wait.Worker) []wait.Worker {
```

Comment:
This function should not be responsible for adding the workers to the caller workers, let's return just the worker(s) and use the `append` in the caller

Reply:
Changed to append the workers in the caller func